ws/src/metadata.rs

```diff
@@ -78,10 +78,12 @@ pub struct RequestContext {
 impl RequestContext {
 	/// Get this session as a `Sink` spawning a new future
 	/// in the underlying event loop.
-	pub fn sender(&self) -> mpsc::Sender<String> {
+	pub fn sender(&self) -> mpsc::UnboundedSender<String> {
+		use futures03::{StreamExt, TryStreamExt};
 		let out = self.out.clone();
-		let (sender, receiver) = mpsc::channel(1);
-		self.executor.spawn(SenderFuture(out, receiver));
+		let (sender, receiver) = mpsc::unbounded();
```

Comment:
The channel was used incorrectly, since it was `cloned` every time before issuing `send`. The spec says that `1` is a reserved value, but there is also at least 1 additional entry for every clone of `Sender`. So basically the channel could never really be full, since each `send` was extending it anyway. I think it could be set to `0` here, but possible it panics, so it's just set to the minimal possible value.